tests/tests.go

```diff
@@ -45,7 +48,15 @@ func StartServerWithContext(t testing.TB, cfg *serverconfig.Config, serverCtx *r
 		serverDone <- serverCtx.Run(ctx, cfg)
 	}()
 
-	testutils.EnsureServiceHealthy(t, cfg.GRPC.Addr, cfg.HTTP.Addr, nil, false)
+	err := testutils.EnsureServiceHealthy(t, cfg.GRPC.Addr, cfg.HTTP.Addr, nil, false)
+	if err != nil {
+		return func() {
```

Comment:
Can't we do this, and the closer/canceller below more elegant with `t.Cleanup`? Then that would allow us to use `require.Error` elsewhere and rest assured that we still get the cleanup invocations at the end of the test. `t.Cleanup` could stop the server and performance any graceful cancellations as needed.

Reply:
I like that idea, but i'd to do it in a separate PR because i'd have to change existing `defers` to `t.cleanups` everywhere (e.g. `defer goleak.VerifyNone`)

Reply:
😓 phew, nice work!